client/components/List/List.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { PropsWithChildren } from 'react';
+
+interface Props {
+  variant?:
+  | 'none'
+  | 'with--bullets'
+  | 'with--checkmark'
+  | 'with--count',
+  type: 'ul' | 'ol',
```

Comment:
Let's rename this to `as` so we are consistent with other Components (eg: `FormattedText`)